test/unit/transloadit/test_smart_cdn.rb

```diff
@@ -0,0 +1,65 @@
+require "test_helper"
+
+describe Transloadit do
+  before do
+    @transloadit = Transloadit.new(key: "my-key", secret: "my-secret")
+    @workspace = "my-app"
+    @template = "test-smart-cdn"
+    @input = "inputs/prinsengracht.jpg"
+    @expire_at = 1732550672867
+  end
+
+  describe "#signed_smart_cdn_url" do
+    it "requires workspace" do
+      assert_raises ArgumentError, "workspace is required" do
+        @transloadit.signed_smart_cdn_url(
+          workspace: nil,
+          template: @template,
+          input: @input
+        )
+      end
+    end
+
+    it "requires template" do
+      assert_raises ArgumentError, "template is required" do
+        @transloadit.signed_smart_cdn_url(
+          workspace: @workspace,
+          template: nil,
+          input: @input
+        )
+      end
+    end
+
+    it "requires input" do
+      assert_raises ArgumentError, "input is required" do
```

Comment:
Internally we have discusses use cases where customers might leave `${fields.input}` unset because they don't need it at all or prefer using named query parameters. When a customer wants to sign such a URL, should we require them to pass `nil` and an explicit empty string to the signature method? I prefer an empty string, which would also work with this implementation.